script/mafia

```diff
@@ -41,7 +43,11 @@ exec_mafia () {
       MAFIA_TEMP=$(mktemp -d 2>/dev/null || mktemp -d -t 'exec_mafia')
       trap "rm -rf \"$MAFIA_TEMP\"" EXIT
 
-      echo "Building mafia-$MAFIA_VERSION in $MAFIA_TEMP"
+      # Create a temporary file in MAFIA_BIN so we can do an atomic copy/move dance.
+      MAFIA_PATH_TEMP=$(mktemp --tmpdir=$MAFIA_BIN $MAFIA_FILE-XXXXXX 2>/dev/null || TMPDIR=$MAFIA_BIN mktemp -t $MAFIA_FILE)
+      trap "rm -f \"$MAFIA_PATH_TEMP\"" EXIT
```

Comment:
yeah that would be bad, it should probably be a single trap to a function call, will fix it up